common/src/main/java/io/druid/collections/ReferenceCountingResourceHolder.java

```diff
@@ -64,6 +64,10 @@ public static long leakedResources()
     return new ReferenceCountingResourceHolder<>(object, object);
   }
 
+  /**
+   * Returns the resource. If multiple threads are supposed to call this method for the same holder,
```

Comment:
It's useful for multiple threads but is not _only_ for use by multiple threads. Reference counting is potentially useful even within a single thread (although I guess we don't use it for this today). I'd suggest wording like,

```
Returns the resource with an initial reference count of 1. More references can be added by
calling {@link #increment()}.
```

Reply:
Good point. Fixed.